buildSrc/src/main/kotlin/com/google/devtools/ksp/AbsolutePathProvider.kt

```diff
@@ -0,0 +1,18 @@
+package com.google.devtools.ksp
+
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.process.CommandLineArgumentProvider
+import java.io.File
+
+class AbsolutePathProvider(
```

Comment:
I'm not sure I like this name `AbsolutePathProvider`. How about we call it something like `RelativizingPathProvider`?